src/TesApi.Web/BatchScheduler.cs

```diff
@@ -963,25 +954,30 @@ private async Task<CloudTask> ConvertTesTaskToBatchTaskAsync(string taskId, TesT
             //    }
             //}
 
-            var batchExecutionDirectoryPath = GetBatchExecutionDirectoryPath(task);
-            var metricsPath = $"/{batchExecutionDirectoryPath}/metrics.txt";
+            var storageUploadPath = GetStorageUploadPath(task, defaultStorageAccountName);
+            var metricsPath = $"/{storageUploadPath}/metrics.txt";
```

Comment:
"metrics.txt" could be a constant, such as "const string metricFilename", because it's used a lot in the rest of this PR